src/components/interactive-builder/edit-question-modal.component.tsx

```diff
@@ -336,7 +366,64 @@ const EditQuestionModal: React.FC<EditQuestionModalProps> = ({
               />
             ) : null}
 
-            {fieldType !== 'ui-select-extended' && (
+            {questionToEdit.type === 'patientIdentifier' && (
+              <div>
+                <FormLabel className={styles.label}>
+                  {t('searchForBackingPeatientIdentifierType', 'Search for a backing patient identifier type')}
+                </FormLabel>
+                {patientidentifierNameLookupError ? (
+                  <InlineNotification
+                    kind="error"
+                    lowContrast
+                    className={styles.error}
+                    title={t('errorFetchingPatientIdentifierTypes', 'Error fetching patient identifier types')}
+                    subtitle={t('pleaseTryAgain', 'Please try again.')}
+                  />
+                ) : null}
+                {isLoadingpatientidentifierName ? (
+                  <InlineLoading className={styles.loader} description={t('loading', 'Loading') + '...'} />
+                ) : (
+                  <>
+                    <Search
+                      defaultValue={patientidentifierName}
+                      id="identifierLookup"
+                      onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+                        handlePatientIdentifierTypeChange(e.target.value)
+                      }
+                      placeholder={t('searchPatientIdentifier', 'Search using identifier type name or UUID')}
+                      required
+                      size="md"
+                      value={selectedPatientIdentifierType['display']}
+                    />
+                    {(() => {
+                      if (!patientIdentifierTypeToLookup) return null;
+                      if (isLoadingPatientIdentifierType)
+                        return (
+                          <InlineLoading className={styles.loader} description={t('searching', 'Searching') + '...'} />
+                        );
+                      if (patientIdentifierTypes?.length && !isLoadingPatientIdentifierType) {
+                        return (
+                          <ul className={styles.patientIdentifierTypeList}>
+                            {patientIdentifierTypes?.map((identifierType, index) => (
+                              <li
+                                role="menuIdentifierTypeitem"
+                                className={styles.patientIdentifierType}
+                                key={index}
+                                onClick={() => handleIdentifierTypeSelect(identifierType)}
+                              >
+                                {identifierType?.display}
+                              </li>
+                            ))}
+                          </ul>
+                        );
+                      }
+                    })()}
```

Comment:
```suggestion
                  <ComboBox
                    id="patientIdentifierTypeLookup"
                    items={patientIdentifierTypes}
                    itemToString={(item: PatientIdentifierType) => item?.display}
                    onChange={({ selectedItem }: { selectedItem: PatientIdentifierType }) => {
                      handleIdentifierTypeSelect(selectedItem);
                    }}
                    placeholder={t('choosePatientIdentifierType', 'Choose a patient identifier type')}
                    initialSelectedItem={patientIdentifierTypes.find(
                      (patientIdentifierType) =>
                        patientIdentifierType.uuid === questionToEdit.questionOptions.identifierType,
                    )}
```

Does something like this work better?